secret-mode.el

```diff
@@ -27,15 +27,31 @@
 
 ;;; Code:
 
-(defconst secret-mode-table
-  (let ((disptbl (make-display-table)))
-    (dotimes (i 1024)
-      (aset disptbl i
-            (pcase (get-char-code-property i 'general-category)
-              ((or 'Cc 'Cf 'Zs 'Zl 'Zp) nil)
-              ('Lu (vector (make-glyph-code ?▆)))
-              (_   (vector (make-glyph-code ?▃))))))
-    disptbl)
+(eval-and-compile
+  (defconst secret-mode--max-letter-char-code
+    (eval-when-compile
```

Comment:
Similarly, what does `eval-when-compile` do?

Reply:
See the Emacs manual link above.